base/applications/mspaint/winproc.cpp

```diff
@@ -949,7 +949,22 @@ LRESULT CMainWindow::OnCommand(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bH
         {
             if (attributesDialog.DoModal(mainWindow.m_hWnd))
             {
-                imageModel.Crop(attributesDialog.newWidth, attributesDialog.newHeight, 0, 0);
+                if (attributesDialog.m_bBlackAndWhite && !imageModel.IsBlackAndWhite())
+                {
+                    CString strText(MAKEINTRESOURCE(IDS_LOSECOLOR));
+                    CString strTitle(MAKEINTRESOURCE(IDS_PROGRAMNAME));
+                    INT id = MessageBox(strText, strTitle, MB_ICONINFORMATION | MB_YESNOCANCEL);
+                    if (id != IDYES)
+                        break;
+
+                    imageModel.PushBlackAndWhite();
```

Comment:
Just to remember, the "Push" concept is in order to have the Revert functionality available?

Reply:
`PushImageForUndo` uses `PushImageForUndo`.
Yep, it can be reverted by Undo.